fusor-ember-cli/app/controllers/subscriptions/management-application.js

```diff
@@ -45,67 +41,68 @@ export default Ember.Controller.extend(NeedsDeploymentMixin, {
       var url = ('/customer_portal/consumers?=' + ownerKey);
 
-      return new Ember.RSVP.Promise(function (resolve, reject) {
-        if (self.get('isInvalidMgmtAppName')) {
-          self.set('showWaitingMessage', false);
-          self.set('showErrorMessage', true);
-          self.set('errorMsg', newSatelliteName + ' failed to be added. Invalid application name.');
-          reject();
-        } else {
-          request({
-            url: url,
-            type: "POST",
-            data: JSON.stringify({
-              name: newSatelliteName,
-              type: "satellite",
-              facts: {
-                "distributor_version": "sat-6.2",
-                "system.certificate_version": "3.2"
-              }
-            }),
-            headers: {
-              "Accept": "application/json",
-              "Content-Type": "application/json",
-              "X-CSRF-Token": token
+      if (this.get('isInvalidMgmtAppName')) {
+        this.set('showWaitingMessage', false);
+        this.set('showErrorMessage', true);
+        this.set('errorMsg', newSatelliteName + ' failed to be added. Invalid application name.');
+      } else {
+        request({
+          url: url,
+          type: "POST",
+          data: JSON.stringify({
+            name: newSatelliteName,
+            type: "satellite",
+            facts: {
+              "distributor_version": "sat-6.2",
+              "system.certificate_version": "3.2"
             }
-          }).then(function(response) {
-            var newMgmtApp = self.store.createRecord(
-              'management-application',
-              {
-                name: response.name,
-                entitlementCount: 0,
-                id: response.uuid
-              }
-            );
+          }),
+          headers: {
+            "Accept": "application/json",
+            "Content-Type": "application/json",
+            "X-CSRF-Token": token
+          }
+        }).then(response => {
+          var newMgmtApp = this.store.createRecord(
+            'management-application',
+            {
+              name: response.name,
+              entitlementCount: 0,
+              id: response.uuid
+            }
+          );
+
+          this.get('model').addObject(newMgmtApp._internalModel);
+          this.get('sessionPortal').set('consumerUUID', response.uuid);
+          this.get('sessionPortal').save();
+          this.set('upstreamConsumerUuid', response.uuid);
+          this.set('upstreamConsumerName', response.name);
+          this.set('showAlertMessage', true);
+          this.set('showWaitingMessage', false);
+        }).catch(error => {
+          let errorMsg = newSatelliteName + ' failed to be added. ';
+          if (error && error.jqXHR && error.jqXHR.responseJSON && error.jqXHR.responseJSON.displayMessage) {
```

Comment:
Wish there was a better way to do this without library support.  I can't think of one off the top of my head.